test: Allow running only a subset of integration tests by setting SELECTED_TESTS

rbalint opened the pull request:
Upstream test is now run in two halves in Ubuntu when the test infra is heavily loaded:

https://git.launchpad.net/~ubuntu-core-dev/ubuntu/+source/systemd/commit/?h=ubuntu-impish&id=a7f894c42d650e3ec62c88c34bdd693a9dfd8df6

The change may be handy for other, too.

bluca reviewed (approved):
LGTM - but for your upstream change, I would suggest to have a look at the runtime of each test for grouping them. The qemu-only tests are the slow ones, so splitting them evenly would probably yield better results. An example:

```
TEST-01-BASIC:                      SUCCESS  ( 39 s)
TEST-02-UNITTESTS:                  SUCCESS  (226 s)
TEST-03-JOBS:                       SUCCESS  ( 14 s)
TEST-04-JOURNAL:                    SUCCESS  ( 15 s)
TEST-05-RLIMITS:                    SUCCESS  (  3 s)
TEST-06-SELINUX:                    SUCCESS  (  0 s)
TEST-07-ISSUE-1981:                 SUCCESS  (  4 s)
TEST-08-ISSUE-2730:                 SUCCESS  ( 80 s)
TEST-09-ISSUE-2691:                 SUCCESS  ( 74 s)
TEST-10-ISSUE-2467:                 SUCCESS  (  3 s)
TEST-11-ISSUE-3166:                 SUCCESS  ( 79 s)
TEST-12-ISSUE-3171:                 SUCCESS  (  6 s)
TEST-13-NSPAWN-SMOKE:               SUCCESS  (360 s)
TEST-14-MACHINE-ID:                 SUCCESS  ( 85 s)
TEST-15-DROPIN:                     SUCCESS  ( 27 s)
TEST-16-EXTEND-TIMEOUT:             SUCCESS  ( 66 s)
TEST-17-UDEV:                       SUCCESS  (338 s)
TEST-18-FAILUREACTION:              SUCCESS  (  5 s)
TEST-19-DELEGATE:                   SUCCESS  ( 84 s)
TEST-20-MAINPIDGAMES:               SUCCESS  (  5 s)
TEST-22-TMPFILES:                   SUCCESS  (  4 s)
TEST-23-TYPE-EXEC:                  SUCCESS  (  2 s)
TEST-24-CRYPTSETUP:                 SUCCESS  (119 s)
TEST-26-SETENV:                     SUCCESS  (  3 s)
TEST-27-STDOUTFILE:                 SUCCESS  (  2 s)
TEST-28-PERCENTJ-WANTEDBY:          SUCCESS  (  5 s)
TEST-29-PORTABLE:                   SUCCESS  (349 s)
TEST-31-DEVICE-ENUMERATION:         SUCCESS  ( 76 s)
TEST-32-OOMPOLICY:                  SUCCESS  ( 75 s)
TEST-33-CLEAN-UNIT:                 SUCCESS  (  6 s)
TEST-34-DYNAMICUSERMIGRATE:         SUCCESS  (  2 s)
TEST-36-NUMAPOLICY:                 SUCCESS  ( 99 s)
TEST-37-RUNTIMEDIRECTORYPRESERVE:   SUCCESS  (  2 s)
TEST-38-FREEZER:                    SUCCESS  ( 94 s)
TEST-39-EXECRELOAD:                 SUCCESS  (  4 s)
TEST-40-EXEC-COMMAND-EX:            SUCCESS  (  3 s)
TEST-41-ONESHOT-RESTART:            SUCCESS  (  9 s)
TEST-42-EXECSTOPPOST:               SUCCESS  ( 24 s)
TEST-43-PRIVATEUSER-UNPRIV:         SUCCESS  (  0 s)
TEST-44-LOG-NAMESPACE:              SUCCESS  (  2 s)
TEST-46-HOMED:                      SUCCESS  (  2 s)
TEST-47-ISSUE-14566:                SUCCESS  ( 10 s)
TEST-48-START-STOP-NO-RELOAD:       SUCCESS  ( 15 s)
TEST-49-RUNTIME-BIND-PATHS:         SUCCESS  ( 13 s)
TEST-50-DISSECT:                    SUCCESS  (301 s)
TEST-51-ISSUE-16115:                SUCCESS  (  8 s)
TEST-52-HONORFIRSTSHUTDOWN:         SUCCESS  (  2 s)
TEST-53-ISSUE-16347:                SUCCESS  ( 82 s)
TEST-54-CREDS:                      SUCCESS  (  2 s)
TEST-55-OOMD:                       SUCCESS  (  3 s)
TEST-56-EXIT-TYPE:                  SUCCESS  ( 10 s)
TEST-58-REPART:                     SUCCESS  ( 71 s)
TEST-59-RELOADING-RESTART:          SUCCESS  ( 12 s)
```

yuwata commented:
(off topic) https://git.launchpad.net/~ubuntu-core-dev/ubuntu/+source/systemd/commit/?h=ubuntu-impish&id=a7f894c42d650e3ec62c88c34bdd693a9dfd8df6 seems to have a copy-and-paste mistake in comment.
```
+# run upstream system integration tests's 1st part
+
+env SELECTED_TESTS="test/TEST-[3-9]*" debian/tests/upstream
```
The comment should be "2nd part", I guess.

rbalint commented:
> (off topic) https://git.launchpad.net/~ubuntu-core-dev/ubuntu/+source/systemd/commit/?h=ubuntu-impish&id=a7f894c42d650e3ec62c88c34bdd693a9dfd8df6 seems to have a copy-and-paste mistake in comment.
> 
> ```
> +# run upstream system integration tests's 1st part
> +
> +env SELECTED_TESTS="test/TEST-[3-9]*" debian/tests/upstream
> ```
> 
> The comment should be "2nd part", I guess.

Yes, I've fixed it later and also updated the change to use `test/run-integration-tests.sh`

rbalint commented:
> LGTM - but for your upstream change, I would suggest to have a look at the runtime of each test for grouping them. The qemu-only tests are the slow ones, so splitting them evenly would probably yield better results. An example:
> 
> ```
> TEST-01-BASIC:                      SUCCESS  ( 39 s)
> TEST-02-UNITTESTS:                  SUCCESS  (226 s)
> TEST-03-JOBS:                       SUCCESS  ( 14 s)
> TEST-04-JOURNAL:                    SUCCESS  ( 15 s)
> TEST-05-RLIMITS:                    SUCCESS  (  3 s)
> TEST-06-SELINUX:                    SUCCESS  (  0 s)
> TEST-07-ISSUE-1981:                 SUCCESS  (  4 s)
> TEST-08-ISSUE-2730:                 SUCCESS  ( 80 s)
> TEST-09-ISSUE-2691:                 SUCCESS  ( 74 s)
> TEST-10-ISSUE-2467:                 SUCCESS  (  3 s)
> TEST-11-ISSUE-3166:                 SUCCESS  ( 79 s)
> TEST-12-ISSUE-3171:                 SUCCESS  (  6 s)
> TEST-13-NSPAWN-SMOKE:               SUCCESS  (360 s)
> TEST-14-MACHINE-ID:                 SUCCESS  ( 85 s)
> TEST-15-DROPIN:                     SUCCESS  ( 27 s)
> TEST-16-EXTEND-TIMEOUT:             SUCCESS  ( 66 s)
> TEST-17-UDEV:                       SUCCESS  (338 s)
> TEST-18-FAILUREACTION:              SUCCESS  (  5 s)
> TEST-19-DELEGATE:                   SUCCESS  ( 84 s)
> TEST-20-MAINPIDGAMES:               SUCCESS  (  5 s)
> TEST-22-TMPFILES:                   SUCCESS  (  4 s)
> TEST-23-TYPE-EXEC:                  SUCCESS  (  2 s)
> TEST-24-CRYPTSETUP:                 SUCCESS  (119 s)
> TEST-26-SETENV:                     SUCCESS  (  3 s)
> TEST-27-STDOUTFILE:                 SUCCESS  (  2 s)
> TEST-28-PERCENTJ-WANTEDBY:          SUCCESS  (  5 s)
> TEST-29-PORTABLE:                   SUCCESS  (349 s)
> TEST-31-DEVICE-ENUMERATION:         SUCCESS  ( 76 s)
> TEST-32-OOMPOLICY:                  SUCCESS  ( 75 s)
> TEST-33-CLEAN-UNIT:                 SUCCESS  (  6 s)
> TEST-34-DYNAMICUSERMIGRATE:         SUCCESS  (  2 s)
> TEST-36-NUMAPOLICY:                 SUCCESS  ( 99 s)
> TEST-37-RUNTIMEDIRECTORYPRESERVE:   SUCCESS  (  2 s)
> TEST-38-FREEZER:                    SUCCESS  ( 94 s)
> TEST-39-EXECRELOAD:                 SUCCESS  (  4 s)
> TEST-40-EXEC-COMMAND-EX:            SUCCESS  (  3 s)
> TEST-41-ONESHOT-RESTART:            SUCCESS  (  9 s)
> TEST-42-EXECSTOPPOST:               SUCCESS  ( 24 s)
> TEST-43-PRIVATEUSER-UNPRIV:         SUCCESS  (  0 s)
> TEST-44-LOG-NAMESPACE:              SUCCESS  (  2 s)
> TEST-46-HOMED:                      SUCCESS  (  2 s)
> TEST-47-ISSUE-14566:                SUCCESS  ( 10 s)
> TEST-48-START-STOP-NO-RELOAD:       SUCCESS  ( 15 s)
> TEST-49-RUNTIME-BIND-PATHS:         SUCCESS  ( 13 s)
> TEST-50-DISSECT:                    SUCCESS  (301 s)
> TEST-51-ISSUE-16115:                SUCCESS  (  8 s)
> TEST-52-HONORFIRSTSHUTDOWN:         SUCCESS  (  2 s)
> TEST-53-ISSUE-16347:                SUCCESS  ( 82 s)
> TEST-54-CREDS:                      SUCCESS  (  2 s)
> TEST-55-OOMD:                       SUCCESS  (  3 s)
> TEST-56-EXIT-TYPE:                  SUCCESS  ( 10 s)
> TEST-58-REPART:                     SUCCESS  ( 71 s)
> TEST-59-RELOADING-RESTART:          SUCCESS  ( 12 s)
> ```

Yes, the first part takes a bit longer, but I also expect that new tests will be added at the end thus this split will work OK for a while and instead of re-balancing a new round should be started when `upstream-2` starts timing out.
